File: spring-java/src/main/java/com/jmpark/springjava/service/TestService.java
package com.jmpark.springjava.service;

import com.jmpark.springjava.entity.District;
import com.jmpark.springjava.repository.DistrictRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class TestService {
    private final DistrictRepository districtRepository;

    public List<District> findAllDistrict() {
        return districtRepository.all();
    }

    public Page<District> pageDistrict(int page, int size) {
        return districtRepository.page(PageRequest.of(page, size));
    }
}

File: spring-kotlin/src/main/kotlin/com/jmpark/springkotlin/service/TestService.kt
package com.jmpark.springkotlin.service

import com.jmpark.springkotlin.entity.District
import com.jmpark.springkotlin.repository.DistrictRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service

@Service
class TestService (
    private val districtRepository: DistrictRepository
) {
    fun findAllDistrict(): List<District> {
        return districtRepository.all()
    }

    fun pageDistrict(page: Int, size: Int): Page<District> {
        return districtRepository.page(PageRequest.of(page, size))
    }
}